frontend/src/App.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {useEffect, useState} from 'react';
+import './css/App.css';
+import DiagramTable from "./components/OverviewPage/DiagramTable";
+import AddForm from "./components/AddPage/AddForm";
+import {AppBar, createTheme, Tab, ThemeProvider, Toolbar, Typography} from "@mui/material";
+import TabContext from '@mui/lab/TabContext';
+import TabList from '@mui/lab/TabList';
+import TabPanel from '@mui/lab/TabPanel';
+import DiagramDetails from "./components/DetailsPage/DiagramDetails";
+import {BpmnDiagramModel} from "./model/BpmnDiagramModel";
+import axios from "axios";
+
+function App() {
+    const [tab, setTab] = useState<string>("Overview")
+    const [detailedDiagram, setDetailedDiagram] = useState<BpmnDiagramModel>({
+        id: ""
+        , name: "-"
+        , businessKey: "-"
+        , filename: "-"
+        , version: 1
+        , calledProcesses: []
+        , commentText: ""
+        , commentAuthor: ""
+        , customDiagram: true
+    })
+    const [bpmnDiagrams, setBpmnDiagrams] = useState<BpmnDiagramModel[]>([])
+
+    const fetchDiagrams = () => {
+        axios.get("/api/bpmndiagrams?distinct=true")
+            .then(response => response.data)
+            .catch(error => console.error("GET-Error: " + error))
+            .then(setBpmnDiagrams)
+    }
+
+    useEffect(() => {
+        fetchDiagrams()
+    },[])
+
+    const theme = createTheme({
+        palette: {
+            primary: {
+                main: '#fc5d0d',
+            },
+            secondary: {
+                main: '#336571',
+            },
+            background: {
+                default: '#f9ded4',
+            },
+        },
+    })
+
+  return (
+    <ThemeProvider theme={theme}>
+        <header>
+            <AppBar position="sticky">
+                <Typography variant="h2" color="inherit">BPMN-Library</Typography>
+            </AppBar>
+        </header>
+        <main>
+            <TabContext value={tab}>
```

Comment:
Für meinen Geschmack ist diese `App.tsx` etwas voll geworden. Überlege Dir, ob Du nicht diesen gesamten TabContext vielleicht in eine Komponente auslagerst. Auch die States, usw. können dann dort hin verschoben werden.
Hier bleibt dann hauptsächlich die Stildefinition und der Header.